daemon/run.go

```diff
@@ -0,0 +1,319 @@
+		// How many were parsed as valid witnesses.
+		Parsed int `json:"parsed"`
+
+		// How many were dropped because the queue was full.
+		Drops int `json:"drops"`
+
+		// How many errors were encountered.
+		Errors int `json:"errors"`
+
+		// How many were processed successfully.
+		Successes int `json:"successes"`
+	}
+
+	type responseBody struct {
+		Message        string         `json:"message,omitempty"`
+		WitnessDetails witnessDetails `json:"witnessDetails,omitempty"`
+	}
+
+	witDetails := witnessDetails{
+		Parsed:    numWitnessesParsed,
+		Drops:     numWitnessesDropped,
+		Errors:    sampledErrs.TotalCount,
+		Successes: successfulEntries,
+	}
+
+	// If we encountered malformed witnesses or there were errors processing witnesses, return a "Bad Request" status.
+	if err != nil || sampledErrs.TotalCount > 0 {
+		message := "Errors encountered while processing witnesses"
+		if err != nil {
+			message = "Malformed witness encountered"
+			witDetails.Errors++
+		}
+		return *NewHTTPResponse(http.StatusBadRequest,
+			responseBody{
+				Message:        message,
+				WitnessDetails: witDetails,
+			})
+	}
+
+	// If we dropped witnesses, return a "Unprocessable Entity" status.
+	if numWitnessesDropped > 0 {
+		return *NewHTTPResponse(http.StatusUnprocessableEntity,
+			responseBody{
+				Message:        "Not all witnesses were processed",
+				WitnessDetails: witDetails,
+			})
+	}
+
+	// Otherwise, everything went well, so return an "OK" status.
+	return *NewHTTPResponse(http.StatusOK,
+		responseBody{
+			WitnessDetails: witDetails,
+		})
+}
+
+// Creates an API model from a learning session in the Akita back end.
+func createModel(request *http.Request) HTTPResponse {
+	vars := mux.Vars(request)
+
+	// Ensure the request body is JSON-encoded.
+	if httpErr := EnsureJSONEncodedRequestBody(request); httpErr != nil {
+		return *httpErr
+	}
+
+	// Get the service ID and instantiate a learning client.
+	serviceID, httpErr := getServiceID(vars)
+	if httpErr != nil {
+		return *httpErr
+	}
+	learnClient := rest.NewLearnClient(cmdArgs.Domain, cmdArgs.ClientID, serviceID)
+
+	// Parse the request body. Expect a JSON serialization of the following type.
+	var requestBody struct {
+		LearnSessionNames []string `json:"learnSessions"`
+	}
+	jsonDecoder := json.NewDecoder(request.Body)
+	if err := jsonDecoder.Decode(&requestBody); err != nil {
+		return *NewHTTPError(err, http.StatusBadRequest, "Malformed request body")
+	}
+
+	// Convert the learning session names into IDs.
+	learnSessionIDs := make([]akid.LearnSessionID, len(requestBody.LearnSessionNames))
+	for i, learnSessionName := range requestBody.LearnSessionNames {
+		var err error
+		learnSessionIDs[i], err = util.GetLearnSessionIDByName(learnClient, learnSessionName)
+		if err != nil {
+			return *NewHTTPError(err, http.StatusNotFound, "Learning session not found: "+learnSessionName)
+		}
+	}
+
+	// Create a new API model with a random name.
+	// TODO: Allow user to specify a name?
```

Comment:
Also good to have, but not necessary for this PR.

Reply:
Done in 2231328.